lib/std/os/bits/linux/bpf.zig

```diff
@@ -0,0 +1,606 @@
+    /// userspace bpf_func_info size
+    func_info_rec_size: u32,
+    func_info: u64,
+    /// number of bpf_func_info records
+    func_info_cnt: u32,
+    /// userspace bpf_line_info size
+    line_info_rec_size: u32,
+    line_info: u64,
+    /// number of bpf_line_info records
+    line_info_cnt: u32,
+    /// in-kernel BTF type id to attach to
+    attact_btf_id: u32,
+    /// 0 to attach to vmlinux
+    attach_prog_id: u32,
+};
+
+/// struct used by Cmd.obj_* commands
+pub const ObjAttr = extern struct {
+    pathname: u64,
+    bpf_fd: fd_t,
+    file_flags: u32,
+};
+
+/// struct used by Cmd.prog_attach/detach commands
+pub const ProgAttachAttr = extern struct {
+    /// container object to attach to
+    target_fd: fd_t,
+    /// eBPF program to attach
+    attach_bpf_fd: fd_t,
+    attach_type: u32,
+    attach_flags: u32,
+    // TODO: BPF_F_REPLACE flags
+    /// previously attached eBPF program to replace if .replace is used
+    replace_bpf_fd: fd_t,
+};
+
+/// struct used by Cmd.prog_test_run command
+pub const TestAttr = extern struct {
+    prog_fd: fd_t,
+    retval: u32,
+    /// input: len of data_in
+    data_size_in: u32,
+    /// input/output: len of data_out. returns ENOSPC if data_out is too small.
+    data_size_out: u32,
+    data_in: u64,
+    data_out: u64,
+    repeat: u32,
+    duration: u32,
+    /// input: len of ctx_in
+    ctx_size_in: u32,
+    /// input/output: len of ctx_out. returns ENOSPC if ctx_out is too small.
+    ctx_size_out: u32,
+    ctx_in: u64,
+    ctx_out: u64,
+};
+
+/// struct used by Cmd.*_get_*_id commands
+pub const GetIdAttr = extern struct {
+    id: extern union {
+        start_id: u32,
+        prog_id: u32,
+        map_id: u32,
+        btf_id: u32,
+        link_id: u32,
+    },
+    next_id: u32,
+    open_flags: u32,
+};
+
+/// struct used by Cmd.obj_get_info_by_fd command
+pub const InfoAttr = extern struct {
+    bpf_fd: fd_t,
+    info_len: u32,
+    info: u64,
+};
+
+/// struct used by Cmd.prog_query command
+pub const QueryAttr = extern struct {
+    /// container object to query
+    target_fd: fd_t,
+    attach_type: u32,
+    query_flags: u32,
+    attach_flags: u32,
+    prog_ids: u64,
+    prog_cnt: u32,
+};
+
+/// struct used by Cmd.raw_tracepoint_open command
+pub const RawTracepointAttr = extern struct {
+    name: u64,
+    prog_fd: fd_t,
+};
+
+/// struct used by Cmd.btf_load command
+pub const BtfLoadAttr = extern struct {
+    btf: u64,
+    btf_log_buf: u64,
+    btf_size: u32,
+    btf_log_size: u32,
+    btf_log_level: u32,
+};
+
+pub const TaskFdQueryAttr = extern struct {
+    /// input: pid
+    pid: pid_t,
+    /// input: fd
+    fd: fd_t,
+    /// input: flags
+    flags: u32,
+    /// input/output: buf len
+    buf_len: u32,
+    /// input/output:
+    ///     tp_name for tracepoint
+    ///     symbol for kprobe
+    ///     filename for uprobe
+    buf: u64,
+    /// output: prod_id
+    prog_id: u32,
+    /// output: BPF_FD_TYPE
+    fd_type: u32,
```

Comment:
looks like this should be an enum? https://elixir.bootlin.com/linux/latest/source/include/uapi/linux/bpf.h#L4181